src/store-api/src/storage/descriptors.rs

```diff
@@ -21,10 +23,83 @@ use crate::storage::{consts, ColumnDefaultConstraint, ColumnSchema, ConcreteData
 pub type ColumnId = u32;
 /// Id of column family, unique in each region.
 pub type ColumnFamilyId = u32;
+/// Sequence number of regions under the same table.
+pub type RegionNumber = u32;
+/// Id of table.
+pub type TableId = u32;
+
 /// Id of the region. It's generated by concatenating table id and region number.
-pub type RegionId = u64;
+#[derive(Default, Clone, Copy, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct RegionId(u64);
 
-pub type RegionNumber = u32;
+impl RegionId {
+    /// Construct a new [RegionId] from table id and region number.
+    pub const fn new(table_id: TableId, region_number: RegionNumber) -> RegionId {
+        RegionId(((table_id as u64) << 32) | region_number as u64)
+    }
+
+    /// Returns the table id of the region.
+    pub const fn table_id(&self) -> TableId {
+        (self.0 >> 32) as TableId
+    }
+
+    /// Returns the region number of the region.
+    pub const fn region_number(&self) -> RegionNumber {
+        self.0 as RegionNumber
+    }
+
+    /// Returns the region id as u64.
+    pub const fn as_u64(&self) -> u64 {
```

Comment:
I suggest rename `as_u64` and `from_u64` to `as_raw` and `from_raw`, as u64 is the raw form of RegionId. `as_u64` and `from_u64` somehow still expose the inner mechanism of RegionId, and a little confusion.